affinity/cpp-23/dXXXXr0.md

```diff
@@ -0,0 +1,295 @@
+keeping threads close to the initial thread, or even avoid migrating threads and
+maintain first touch policy. Nevertheless, most programs can change their work
+distribution, especially in the presence of nested parallelism.
+
+Frequently, data are initialized at the beginning of the program by the initial
+thread and are used by multiple threads. While some OSes automatically migrate
+threads or data for better affinity, migration may have high overhead. In an
+optimal case, the OS may automatically detect which thread access which data
+most frequently, or it may replicate data which are read by multiple threads, or
+migrate data which are modified and used by threads residing on remote locality
+groups. However, the OS often does a reasonable job, if the machine is not
+overloaded, if the application carefully uses first-touch allocation, and if the
+program does not change its behavior with respect to locality.
+
+The affinity interface we propose should help computers achieve a much higher
+fraction of peak memory bandwidth when using parallel algorithms. In the future,
+we plan to extend this to heterogeneous and distributed computing. This follows
+the lead of OpenMP [[3]][design-of-openmp].
+
+# Proposal
+
+This paper proposes a new group of properties (as described in P1393r0
+[[4]][p1393r0] that allow users to specify how a callable in a bulk algorithm
+such as the `bulk_execute` algorithm in [[2]][p0443r12] will perform in light of
+the placement of the work-items it invokes. This information can then be used by
+an an executor implementation which is aware of the property to efficiently map
+the work-items it invokes to the underlying resources which it represents.
+
+## Shift to descriptive properties
+
+The previous revision of this proposal; P1436r3 [[1]][p1436r3] took a
+prescriptive approach. It suggested the `bulk_execution_affinity`
+properties, which targeted executors, providing a hint to the executor
+implementation that it should aim to map the work-items invoked by a bulk
+algorithm to the underlying resources of the executor in scatter or gather
+pattern to achieve optimal performance for the callable being invoked.
+
+This proposal switches perspective on this, and instead takes a
+descriptive approach. It suggests the `adjacency` properties, which instead
+target execution policies, providing a hint to the executor implementation which
+expresses the performance implications of the callable on the mapping of the
+work-items invoked by a bulk algorithm, allowing the implementation to choose
+the appropriate action.
+
+This shift moves towards a more algorithmic-based approach to the customization
+of execution and both removes the need for users to understand the internal
+workings of various executors and allows more freedom in executor
+implementations to choose how to interpret the hint.
+
+The properties of the previous revision of this paper; `bulk_execution_affinity`
```

Comment:
"... of this paper, `bulk_execution_affinity`, ..."

Reply:
Done.